bundles/go/go_fmt.moon

```diff
@@ -0,0 +1,58 @@
+import app, config, mode, signal from howl
+import Process from howl.io
+
+append = table.insert
+
+run_command = (file) ->
+  args = {}
+  for arg in config.go_fmt_command\gmatch '%S+'
+    append args, arg
+  append args, file
+  success, out, err, p = pcall Process.execute, args
+  unless success
+    log.error out
+    return
+  unless p.successful
+    log.error err
+    return
+
+reload_buffer = (buffer) ->
+  buffer.text = buffer.file.contents
+  buffer.sync_etag = buffer.file.etag
+  buffer.modified = false
+  
+calculate_new_pos = (pos, before, after) ->
+  new_pos = 1
+  i = 1
+  while i <= pos and new_pos <= #after
+    if before[i] == after[new_pos]
+      i += 1
+      new_pos += 1
+    elseif before[i]\match '%s'
+      i += 1
+    else
+      new_pos += 1
+  new_pos-1
+
+fmt = (buffer) ->
+  return unless buffer.mode.name == 'go'
+
+  file = buffer.file
+  before = file.contents
+  run_command file
```

Comment:
read_only = true should prevent typing but see #155 

BTW, I feel saving is such a critical operation that having it depend on an external command is not ideal. In the rare case where `gofmt` breaks or the `Process` machinery errors out, the save could be cancelled and maybe even prevent saving permanently. What do you think about always saving first and then applying 'gofmt` to save again?

Even if in this case it is super reliable, this introduces a pattern that may break if another bundle adds a less reliable `before_save`. I'd prefer to see a very simple and predictable code path between pressing `ctrl_s` and the actual save.